thrust/thrust/iterator/zip_iterator.h

```diff
@@ -195,6 +195,15 @@ class _CCCL_DECLSPEC_EMPTY_BASES zip_iterator : public detail::make_zip_iterator
       : m_iterator_tuple(iterator_tuple)
   {}
 
+  //! This constructor creates a new \p zip_iterator from a pack of iterators.
+  //!
+  //! \param iterators A pack of iterators to zip.
+  template <typename... Iterators,
+            ::cuda::std::enable_if_t<(::cuda::std::input_or_output_iterator<Iterators> && ...), int> = 0>
+  _CCCL_HOST_DEVICE zip_iterator(Iterators... iterators)
+      : m_iterator_tuple(::cuda::std::move(iterators)...)
+  {}
```

Comment:
We should take by forwarding reference I believe
```suggestion
  _CCCL_HOST_DEVICE zip_iterator(Iterators&&... iterators)
      : m_iterator_tuple(::cuda::std::forward<Iterators>(iterators)...)
  {}
```

Reply:
I thought we would be fine, since `make_zip_iterator` also takes the iterators by value and copies them. Also, perfectly forwarding ctors are a hazard sometimes. But I will try.